java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -28,6 +30,59 @@ protected static Object handleObjectResponse(Response response) {
         return (new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer)).apply(response);
     }
 
+    /**
+     * Check for errors in the Response and return null Throws an error if an unexpected value is
+     * returned
+     *
+     * @return null if the response is empty
+     */
+    protected static Void handleVoidResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value == null) {
+            return null;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected null");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a String.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a String
+     */
+    protected static String handleStringResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof String) {
+            return (String) value;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected String");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the * value as a HashMap
+     *
+     * @param response Redis protobuf message
+     * @return Response as a String
+     */
+    protected static HashMap<String, Object> handleMapResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof HashMap) {
+            return (HashMap<String, Object>) value;
```

Comment:
```suggestion
        if (value instanceof Map) {
            return (Map<String, Object>) value;
```
Having a `HashMap` reference here we're adding an extra dependency on the implementaion on rust side.